rpc/dial.go

```diff
@@ -207,12 +214,27 @@ func dial(
 					)
 				}
 				dialCh <- dialResult{conn: conn, cached: cached}
-			case !errors.Is(err, ErrNoWebRTCSignaler):
-				// TODO(RSDK-6493): Investigate if we must `skipDirect` here.
-				dialCh <- dialResult{err: err, skipDirect: true}
 			case ctxParallel.Err() != nil:
+				if dOpts.debug {
+					logger.Debugw(
+						"webrtc dial failed and context finished, will not try to dial directly",
+						"dial error", err,
+						"context error", ctxParallel.Err(),
+					)
+				}
 				dialCh <- dialResult{err: ctxParallel.Err(), skipDirect: true}
+			case errors.Is(err, ErrNoWebRTCSignaler):
+				// There is no WebRTC signaler, so dialing WebRTC is expected to fail.
+				// We should still try to dial directly as a fallback.
+				dialCh <- dialResult{err: err}
+			case isUnauthenticatedError(err):
+				// We are our not authenticated to dial the host, so we should not
+				// attempt to dial directly either.
+				dialCh <- dialResult{err: err, skipDirect: true}
```

Comment:
Some tests break without this check, but they contain scenarios where webrtc dial requires authentication, but direct dial does not.